### PR TITLE
Consolidates some XSS under hivecore

### PR DESCRIPTION
**Pull request by boskoramen**
# About the pull request

Spawn pool and evo pods are removed and their functionality is umbrella'd under the hive core. Sprites still exist though.

# Explain why it's good for the game

"Roleplay" has become an increasingly more popular and touchy subject within the community as of recently. I, wholeheartedly agree, that roleplay is an important aspect of the game, and there are ways to improve it. One of these ways is through immersion. In this MR, it is intended to increase player immersion.

One of the most memorable and haunting scenes in Aliens was when they reached the hive and they found the bodies. This very unique and cinematic scene was often able to be replicated in CM, with the marines busting open the hive and finding all the chestbursted bodies of their comrades. Roleplaying this out was commonplace. "Dear God... what did they do to you..." or acting out in disgust are just a few of the many ways having bodies in the hive positively impacted the game. 

XSS was a failed attempt at spicing up the xenomorph gameplay loop at the expense of immersion and should be at least somewhat reverted while keeping balance in mind. A brief discussion with many prominent xenomorph players, including those most experienced in queen, have not particularly expressed favor to XSS either.

To start, let us remember why XSS (xeno special structures - hive core, evo pods, morpher, pool, etc) was added.
https://docs.google.com/document/d/19_zDUmLdxpUzxj-GuWu7F4bSj4zBYzmZ39s_N5X7kQ0/edit - This is the original development document for XSS. 
Let us examine the major points:
1.Introduce a way for Xenomorph players to recycle - This idea was never reached. 

2.Reduce Xenomorph attrition - Grand objective was unsuccessful. Very little changed.

3. Offer new avenues of play "by reducing the punishment of xenos dying" - This never happened. Dying was still just as punishing, especially with facehug nerfs.

The spawn pool - The idea of the spawn pool was successful and has remained unchanged since. I would argue, however, it is not immersive. Xenomorphs do not have bright, glowing, acidic pools in their hives. (Yes I know there was a comic with a pool, and this is not how it was used)

Egg Morpher - These used to be TURRETS. They are no longer turrets, and their sprites have been broken for almost 4 years (the bodies put inside of them used to show their face in the little purple part) They are currently defunct facehugger reservoirs. I am in favor of removing them, but I would argue that is a balance issue (number of huggers in play)

Evolution Pod - It was intended for these to be able to be eaten in order to evolve. They haven't done this for years. Why do we still have them? They take up 18 spaces of precious hive weeds, provide light, (xenomorphs HATE light) and wherever a hive core is built, these are also built. We can just merge them with the hive core because there is no reason to have them anymore.

This PR currently completely removes the spawn pool and evolution pod from gameplay, while reimagining their functions for current balance. This PR is not intended to change balance in any way.

All functionality from the spawn pool in regards to "pooled" larva has been given to the hive core, and they are now called "burrowed" larva. 
Chestbursts now give two larva, this is to be kept with current balance of two xenos per capture. 

Evopod functionality and evolution speed boost was merged with the hive core.

# Testing Photographs and Procedure
n/a

# Changelog

:cl: TheDonkified
del: Evo pods and spawn pool are removed.
add: Hive core directly affects evolution speed and is where burrowed larva spawn from now on.
/:cl:

### CODE REVIEW

File: code/modules/cm_aliens/structures/special/pylon_core.dm [hunk 1]
```diff
@@ -127,11 +132,106 @@
...
+	return linked_hive.stored_larva
+
+/obj/effect/alien/resin/special/pylon/core/proc/spawn_burrowed_larva(mob/xeno_candidate)
+	if(can_spawn_larva() && xeno_candidate)
+		var/mob/living/carbon/xenomorph/larva/new_xeno = spawn_hivenumber_larva(loc, linked_hive.hivenumber)
+		if(isnull(new_xeno))
+			return FALSE
+
+		new_xeno.visible_message(SPAN_XENODANGER("A larva suddenly emerges from \the [src]!"),
+		SPAN_XENODANGER("You emerge from \the [src] and awaken from your slumber. For the Hive!"))
```
Comment (harryob) [inline code/modules/cm_aliens/structures/special/pylon_core.dm:176]:
```suggestion
		new_xeno.visible_message(SPAN_XENODANGER("A larva suddenly emerges from [src]!"),
		SPAN_XENODANGER("You emerge from [src] and awaken from your slumber. For the Hive!"))
```

File: code/modules/cm_aliens/structures/special/pylon_core.dm [hunk 2]
```diff
@@ -127,11 +132,106 @@
...
+		msg_admin_niche("[key_name(new_xeno)] emerged from \a [src]. (<A HREF='?_src_=admin_holder;[HrefToken(forceGlobal = TRUE)];adminplayerobservecoodjump=1;X=[src.x];Y=[src.y];Z=[src.z]'>JMP</a>)")
```
Comment (harryob) [inline code/modules/cm_aliens/structures/special/pylon_core.dm:177]:
```suggestion
		msg_admin_niche("[key_name(new_xeno)] emerged from \a [src]. (<A HREF='?_src_=admin_holder;[HrefToken(forceGlobal = TRUE)];adminplayerobservecoodjump=1;X=[x];Y=[y];Z=[z]'>JMP</a>)")
```

File: code/modules/cm_aliens/structures/special/pylon_core.dm [hunk 3]
```diff
@@ -127,11 +132,106 @@
...
+			larva_amount = 5
+		else
+			larva_amount += max(xeno.tier, 1) // Now you always gain larva.
+	else
+		return
+
+	if(!do_after(user, 10, INTERRUPT_ALL|BEHAVIOR_IMMOBILE, BUSY_ICON_GENERIC))
+		return
+
+	visible_message(SPAN_DANGER("\The [src] engulfs \the [xeno] in resin!"))
```
Comment (harryob) [inline code/modules/cm_aliens/structures/special/pylon_core.dm:229]:
```suggestion
	visible_message(SPAN_DANGER("[src] engulfs [xeno] in resin!"))
```

File: code/modules/cm_aliens/structures/special/pylon_core.dm [hunk 4]
```diff
@@ -184,8 +282,11 @@
 			linked_hive.hive_structures_limit[XENO_STRUCTURE_POOL] = 0
 			xeno_announcement("\The [linked_hive.name] has lost their hive core!", "everything", HIGHER_FORCE_ANNOUNCE)
 
-			if(linked_hive.spawn_pool)
-				qdel(linked_hive.spawn_pool)
+		if(linked_hive.hijack_burrowed_surge)
+			visible_message(SPAN_XENODANGER("You hear something resembling a scream from [src] as it's destroyed!"))
+			xeno_message(SPAN_XENOANNOUNCE("Psychic pain storms throughout the hive as \the [src] is destroyed! You will no longer gain burrowed larva over time."), 3, linked_hive.hivenumber)
```
Comment (harryob) [inline code/modules/cm_aliens/structures/special/pylon_core.dm:287]:
```suggestion
			xeno_message(SPAN_XENOANNOUNCE("Psychic pain storms throughout the hive as [src] is destroyed! You will no longer gain burrowed larva over time."), 3, linked_hive.hivenumber)
```

**Review by harryob — CHANGES_REQUESTED** [review]
just some minor things for the sake of standards while you're there

**Review by morrowwolf — CHANGES_REQUESTED** [review]
Doubled bursts at roundstart. Please fix.

**Comment by morrowwolf** [comment]
This is cool so I fixed it for ya. Let's see how next test goes.

**Review by morrowwolf — APPROVED** [review]
Code seems fine, no issues during TM. I enjoy a more focused looking hive and from time to time (when there aren't a dozen egg morphers) having bodies to actually come and recover when breaching the hive.

**Pull request merged**